resources/idlharness.js

```diff
@@ -212,44 +212,56 @@ IdlArray.prototype.add_untested_idls = function(raw_idls, options)
 {
     /** Entry point.  See documentation at beginning of file. */
     var parsed_idls = WebIDL2.parse(raw_idls);
-    for (var i = 0; i < parsed_idls.length; i++)
-    {
+    this.mark_as_untested(parsed_idls);
+    this.internal_add_idls(parsed_idls, options);
+};
+
+//@}
+IdlArray.prototype.mark_as_untested = function (parsed_idls)
+//@{
+{
+    for (var i = 0; i < parsed_idls.length; i++) {
         parsed_idls[i].untested = true;
-        if ("members" in parsed_idls[i])
-        {
-            for (var j = 0; j < parsed_idls[i].members.length; j++)
-            {
+        if ("members" in parsed_idls[i]) {
+            for (var j = 0; j < parsed_idls[i].members.length; j++) {
                 parsed_idls[i].members[j].untested = true;
             }
         }
     }
-    this.internal_add_idls(parsed_idls, options);
 };
+//@}
+
+//@}
+IdlArray.prototype.is_excluded_by_options = function (name, options)
+//@{
+{
+    return !name ||
+        options && (
+            options.except && options.except.includes(name) ||
+            options.only && !options.only.includes(name));
+};
+//@}
 
 //@}
 IdlArray.prototype.add_dependency_idls = function(raw_idls, options)
 //@{
 {
     const parsed_idls = WebIDL2.parse(raw_idls);
     const new_options = { only: [] }
-    
+
     const all_deps = new Set();
-    Object.values(this["inheritance"]).forEach(v => all_deps.add(v));
-    Object.values(this["implements"]).forEach(v => all_deps.add(v));
+    Object.values(this.inheritance).forEach(v => all_deps.add(v));
+    Object.values(this.implements).forEach(v => all_deps.add(v));
+    Object.values(this.includes).forEach(v => all_deps.add(v));
     this.partials.map(p => p.name).forEach(v => all_deps.add(v));
-    
-    console.log('all deps', all_deps);
 
     if (options && options.except && options.only) {
         throw new IdlHarnessError("The only and except options can't be used together.");
     }
+
     const skip = name => {
```

Comment:
Can you name this `should_skip` like the other one?

Reply:
Done.